### DISCUSSION TITLE
LED as Light-Sensor (Cheap)

### DISCUSSION THREAD

**Discussion by freifunkerffm** [arendst/Tasmota#17193] [Ideas]
Hi all, 

I came across the possibility that a simple LED could be (miss)used as a cheap sensor for light (At least it reports is it light or dark)
Original post:
https://playground.arduino.cc/Learning/LEDSensor/

I did not found a kind of "Tutorial/Howto" that explains in which way a feature can be added to TASMOTA and I am not that good in reengeneering/coding.
But I see here the following tasks:
- Add this LED-Light Sensor as a new feature option in TASMOTA
- If this Feature is set: Add two new "Pin-Action" in TASMOTA "Module parameters" config page (One for LED-Light Sensor Anode and one for Cathode)
- If this Feature is set: Trigger the code in TASMOTA to trigger the check and reply back the time it took to have the Pin back to logical 0 via MQTT, that is indicating if it is light or dark.

What do you think, is this may a useful and valid TASMOTA feature?

So long

Kai

**Comment by barbudor** [comment]
Totally useless waste of time and flash to implement that in Tasmota
There are already good light sensors solutions available for Tasmota from basic LDR on analog input to great I2C sensors with measurment of visible, IR and UV
Pickup one that suits your need.
